linked_list/linked_list.py

```diff
@@ -21,15 +24,52 @@ class LinkedList:
   def __init__(self):
     self.head = None
     self.tail = None
+#we make methods to interact with the node class
 
+#add_to_tail
+#"This involves pointing the next pointer of the the current last node of the linked list
+#to the new data node. So the current last node of the linked list becomes the second last 
+# data node and the new node becomes the last node of the linked list."
   def add_to_tail(self, value):
-    pass
+    if self.head == None and self.tail == None:
+      self.head = Node(value)
+      self.tail = Node(value)
+    else:
+      if self.head.value == None and self.tail.value == None:
+        self.head = Node(value)
+        self.tail = Node(value)
+      else:
+        if self.head.next_node == None:
+          self.head.set_next(value)
```

Comment:
Remember that this is an add_to_tail method, so you should be adding a node after the the current tail. Also, you're passing in a number here. The next_node of a node should be a node type object, i.e., 

```
self.head.set_next(Node(value))
```